### ISSUE TITLE
Model containing 'Canvas View'/'Scetch View' with reference to 'ArchiMate View' export fial.

### ISSUE THREAD

**Issue by ZinovyevMaxim** [archi-contribs/database-plugin#6]
Hi,
I'm sorry for my very bad english language...

I'm try export model containing 'Scetch View' and 'Canvas View' with refernce link on 'ArchiMate View'. Export fail with error "Don't know how to save CanvasModel child: <name ArchiMate View>" (DiagramModelReference)... Why ??? In 'Canvas view' contained tools (arrow, images, blocks, sticky) advanced visual description for interaction between 'ArchiMate View' on schema.

Thanks!

**Comment by herve91** [comment]
Could you please confirm t he version of the plugin you are using ?

Thanks
Hervé

**Comment by ZinovyevMaxim** [comment]
I'm trying export for 3 and 4 versions. 4 version refernce link export for 'ArchiMate View', not exported for Canvas View and ScetchView

In plugin for 3 version Archi MySQL database script for create database not work with reference link  to diagram table not contained column "diagrammodelid"

v0.12 (24/10/2016)
Add column "diagrammodelid" to table "diagrammodelreference"

**Comment by ZinovyevMaxim** [comment]
Archi 3, plugin for Archi 3 1.0, MySQL
1.export reference link to other view in ArchiMate View - not work (script for creation not contained change v0.12 (24/10/2016))
2.export reference link to other view in Canvas View - not work
3.export reference link to other view in Scetch View - not work
Archi 4, plugin for Archi 4 1.0, MySQL
1.export reference link to other view in ArchiMate View - work
2.export reference link to other view in Canvas View - not work
3.export reference link to other view in Scetch View - not work

**Comment by herve91** [comment]
Ok, I'll have a look tonight when I come back home and I let you know asap ...

Best regards
Hervé

**Comment by herve91** [comment]
I succeeded to reproduct the behaviour you explained.

I had effectively forgotten these two cases. They were missing from my test use cases. Thanks a lot for spotting them.

I uploaded new release 1.0.1.

**Comment by ZinovyevMaxim** [comment]
Test for Archi 4, MySQL, plugin v 1.0.1 - OK
For Archi 3 I'm not see correct script for creation table. Last update in repository 3 month ago (v0.12 (24/10/2016))
create tables (MySQL).sql	3 months ago

Thanks for operative correct.

P.S. One offtop question: when planed realised link between elements standalone models.

**Comment by herve91** [comment]
Got it. I just uploaded corrected files with missing column. Thanks again.

Regarding the link, it is too complicated to implement regarding the way I developped the version 1.x of the plugin. Instead I started a brand new version 2.x that will work differently: all the element and relationships will be stored in a common zone, distinct from the models. It will be possible to import any element/relationship in any model. If the element/relationship is updated in one model, this update will be visible in all the other models.

I updated the "roadmap" page of my wiki and will provide more information soon :)
